driver/sources/src/main/scala/org/virtuslab/ideprobe/config/PathsConfig.scala

```diff
@@ -0,0 +1,26 @@
+package org.virtuslab.ideprobe.config
+
+import java.nio.file.{Path, Paths}
+import org.virtuslab.ideprobe.config.PathsConfig.{cachePath, screenshotsPath, tmpDir}
+
+case class PathsConfig(instances: Path = tmpDir,
```

Comment:
Something is wrong in this file. Why instances and worksaces by default go directly to temp dir?
Btw as I shown in the issue in example code:
```
val targetCloneDir = fixture.paths.base.resolve("pants-repos").create().resolve(Hash.md5("$uri-$ref"))
```
The base path for probe paths should also be available in the code so that extensions can use it to create its files in the same place as core code.

Also note that we often have a separation of `Config` objects and domain objects in form of for example IntelliJFactory that has a method `from` (btw. probably we should rename all of these methods to `fromConfig`) that creates itself i.e. the domain object from config object. I think we should do a similar thing here. Have IdeProbePaths/TemporaryPaths or something and create it from PathsConfig. This is useful because of another thing I'd like to have:
someone can just say `probe.paths.base = "/mnt/gfs/ideprobe"` in their config, and all of the directories that depend on base path will go to a subdir of that one. If someone changes `probe.paths.cache = "/Users/uwawrzyk/.cache/ideprobe"`, the cache would go to my home dir, but other things will still go where the base dir says (tmp). This can be implemented easily if you make all of the in PathsConfig an Option[Path], and then the domain object would not have Options thanks to the mapping logic.